src/components/dataset/schemaCreation/DatasetSchemaInformationView.tsx

```diff
@@ -0,0 +1,334 @@
+          className={clsx(classes.formLabel, { [classes.formLabelError]: errors.description })}
+        >
+          Dataset description*
+        </label>
+        <Controller
+          name="description"
+          control={control}
+          rules={{ required: 'description is required' }}
+          render={({ field }) => (
+            <>
+              <SimpleMdeReact options={editorOptions} {...field} />
+              {errors.description && (
+                <span className={classes.formInputError}>{errors.description.message}</span>
+              )}
+            </>
+          )}
+        />
+      </Grid>
+
+      <Grid item xs={6}>
+        <label
+          htmlFor="dataset-terraProject"
+          className={clsx(classes.formLabel, { [classes.formLabelError]: errors.terraProject })}
+        >
+          Terra project*
+        </label>
+        <Controller
+          name="terraProject"
+          control={control}
+          render={({ field }) => (
+            <Select id="dataset-terraProject" className={classes.formInput} {...field}>
+              <MenuItem value="yes">Yes</MenuItem>
+              <MenuItem value="no">No</MenuItem>
+            </Select>
+          )}
+        />
+      </Grid>
+
+      <Grid item xs={6}>
+        <label htmlFor="dataset-enableSecureMonitoring" className={classes.formLabel}>
+          Secure monitoring
+        </label>
+        <Controller
+          name="enableSecureMonitoring"
+          control={control}
+          render={({ field }) => (
+            <Select id="dataset-enableSecureMonitoring" className={classes.formInput} {...field}>
+              <MenuItem value="true">Yes</MenuItem>
+              <MenuItem value="false">No</MenuItem>
+            </Select>
+          )}
+        />
+      </Grid>
+
+      <Grid item xs={6}>
+        <label htmlFor="dataset-cloudPlatform" className={classes.formLabel}>
+          Cloud Platform*
+        </label>
+        <Controller
+          name="cloudPlatform"
+          control={control}
+          rules={{ required: true }}
+          render={({ field }) => (
+            <Select
+              id="dataset-cloudPlatform"
+              className={classes.formInput}
+              {...field}
+              onChange={(event: any, change: any) => {
+                const cloudPlatform = event.target.value;
+                setRegionOptions(cloudPlatform === 'gcp' ? GCP_REGIONS : AZURE_REGIONS);
+                field.onChange(event, change);
+              }}
+            >
+              <MenuItem value="gcp">Google Cloud Platform (GCP)</MenuItem>
+              <MenuItem value="azure">Azure</MenuItem>
+            </Select>
+          )}
+        />
+      </Grid>
+
+      <Grid item xs={6} className={classes.formFieldContainer}>
+        <label
+          htmlFor="dataset-region"
+          className={clsx(classes.formLabel, { [classes.formLabelError]: errors.region })}
+        >
+          Region*
```

Comment:
ok! I wanted this to be different actually, because the cloud platform only has 2 options, so a select is easy. But when the possible regions are > 100, i didn't want the user to have to scroll through the list, and wanted to give them the ability to type the answer in.

I removed the flag `freeSolo` that allowed that component to allow any value (so now it is constrained to the known options list), but still allows the user to type.

Does this help? Let me know if i should readjust!

Reply:
Gotcha! Yes, I like that you can't actually submit an invalid region now.